File: Difficulty: Medium/Next Prime Number/next-prime-number.java
class Solution {
    public static int nextPrime(int n) {
        if (n <= 1) {
            return 2;
        }

        n++;
        while (!isPrime(n)) {
            n++;
        }

        return n;
    }

    public static boolean isPrime(int num) {
        if (num <= 1) return false;
        for (int i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
}

File: Difficulty: Medium/Next Prime Number/next-prime-number.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main() {
    int num;
    cin >> num;

    int res = 0;
    bool prime = false;

    for (int i = num + 1;; ++i) {
        prime = true;
        for (int j = 2; j <= sqrt(i); ++j) {
            if (i % j == 0) {
                prime = false;
                break;
            }
        }
        if (prime) {
            res = i;
            break;
        }
    }

    cout << res << endl;
    return 0;
}
